test/functional/tools/StaticServer.ts

```diff
@@ -1,33 +1,40 @@
-import * as express from 'express';
+import { exec, ChildProcess } from 'child_process';
 
 export class StaticServer {
-  private app: any;
-  private server: any;
+  private proc: ChildProcess;
 
-  constructor(private directory: string, private port: number) {}
+  constructor(private repoRoot: string) {}
 
   public launch(): Promise<void | Error> {
-    this.app = express();
-    this.app.use(express.static(this.directory));
-    return new Promise((resolve, reject) => {
-      this.app.on('error', (err: Error) => {
-        reject(err);
-      });
-      this.server = this.app.listen(this.port, () => {
-        resolve();
+    return new Promise(resolve => {
+      this.proc = exec(
+        './node_modules/.bin/webpack-dev-server --mode=production',
+        {
+          cwd: this.repoRoot,
+          env: {
+            PATH: process.env.PATH,
+            NODE_OPTIONS: '--max_old_space_size=8192',
+          },
+        },
+        (err, _stdout, _stderr) => {
+          if (err != null) {
+            reject(err);
+          }
+        },
+      );
+      this.proc.stdout.on('data', (data: Buffer) => {
+        const str = String(data);
+        if (str.match(/compiled/i)) {
```

Comment:
Not convinced this is sound but I couldn't think of another way of detecting when the dev server was ready to serve up the build.

Reply:
I've used `wait-on` npm package to wait for 200 responses.  

https://github.com/jeffbski/wait-on

see https://github.com/ui-router/cypress-runner/blob/05116d8b5782025a2ddf1f6c4b6423235a20e9af/cypress-runner.js#L73-L84